// Copyright 2020 Google LLC. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package tfx.orchestration;

import "ml_metadata/proto/metadata_store.proto";

// Message that represents a list of artifacts.
message ArtifactList {
  repeated ml_metadata.Artifact artifacts = 1;
}

// All the information needed for a TFX component execution.
message ExecutorInvocation {
  map<string, ArtifactList> inputs = 1;
  map<string, ArtifactList> outputs = 2;
  map<string, ml_metadata.Value> execution_properties = 3;
  string output_metadata_uri = 4;
}
